content/post/gld.md

```diff
@@ -0,0 +1,1355 @@
+|[google/licenseclassifier](https://github.com/google/licenseclassifier)| 76% \\(\\quad(\\frac{682}{902})\\) | 907 |
+|[boyter/lc](https://github.com/boyter/lc)| 88% \\(\\quad(\\frac{797}{902})\\) | 548 |
+
+The total number of repositories in the dataset is 958, however, only 902 contain any pointer to
+the license - we looked though each of them. The rest are mainly "awesome lists" and Chinese projects
+and translations of the western books. We filed issues for the maintainers to clarify the license
+[\[1\]](https://github.com/DrkSephy/es6-cheatsheet/issues/90)
+[\[2\]](https://github.com/kdn251/interviews/issues/63)
+[\[3\]](https://github.com/markerikson/react-redux-links/issues/87)
+[\[4\]](https://github.com/CodeHubApp/CodeHub/issues/441)
+[\[5\]](https://github.com/h4cc/awesome-elixir/issues/4400)
+[\[6\]](https://github.com/fffaraz/awesome-cpp/issues/449).
+Few added the license; some maintainers even demonstrated aggression.
+We also encountered two licenses which were not included into SPDX and reported them:
+[\[1\]](https://github.com/spdx/license-list-XML/issues/611)
+[\[2\]](https://github.com/spdx/license-list-XML/issues/612).
+
+#### How we measured the time
+
+Hardware: Intel Core i7-7500U (2x2 threads), 2x8GB LPDDR3@1867MHz.
+
+```bash
+$ cd $(go env GOPATH)/src/gopkg.in/src-d/go-license-detector.v2/licensedb
+$ mkdir dataset && cd dataset
+$ unzip ../dataset.zip
+$ # src-d/go-license-detector
+$ time license-detector * \
+  | grep -Pzo '\n[-0-9a-zA-Z]+\n\tno license' | grep -Pa '\tno ' | wc -l
+$ # benbalter/licensee
+$ time ls -1 | xargs -n1 -P4 licensee \
+  | grep -E "^License: Other" | wc -l
+$ # google/licenseclassifier
+$ time find -type f -print | xargs -n1 -P4 identify_license \
+  | cut -d/ -f2 | sort | uniq | wc -l
+$ # boyter/lc
+$ time lc . \
+  | grep -vE 'NOASSERTION|----|Directory' | cut -d" " -f1 | sort | uniq | wc -l
+```
+
+## Algorithm
+
+We have implemented license detection based on the license and README files for now, and wish to
+add fine-grained scanning of source code files in the future (do you want to help us?). Given the
+stated license text, we compare it to the texts in the SPDX database and record the match. The naive
+way of making the comparisons is to calculate the
+[Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance)
+between the query and each of the examples. This is not fast at all for the two reasons:
+
+1. The complexity is linear to the number of examples.
+2. The complexity is linear to the size of the difference. Since we've always got very few matches,
+the rest of the distances will be slow to calculate.
+
+These two reasons confidently render the naive approach unusable.
+
+The core of go-license-detector's detection mechanism is [Locality Sensitive Hashing](https://en.wikipedia.org/wiki/Locality-sensitive_hashing).
+We treat each license as a TF-IDF-weighted bag-of-words - that is, as a set of words where every
```

Comment:
Replaced `TF-IDF` with its full name, since you only use it once
And link to https://en.wikipedia.org/wiki/Tf%E2%80%93idf